homeassistant/components/google_assistant/report_state.py

```diff
@@ -80,6 +80,10 @@ async def async_entity_state_listener(
         ):
             return
 
+        # For doorbell events we use notitication and we need to synchronize
+        # them using a `SYNC` response together with other state changes.
+        # See: https://developers.home.google.com/cloud-to-cloud/guides/doorbell#objectdetection
+        # We do not support a combined camera stream.
         if (notifications := entity.notifications_serialize()) is not None:
```

Comment:
At the moment if a doorbell event have attributes, that change, that will trigger a notification event if it's within 30 seconds of previous event. We need to check old state and verify there was a change in state, not just an attribute. 

We could pass along the old_state here to verify, or we could cache notifications already done, and not generate new notifications if already sent.

Reply:
You are right, but practically I would not expect attributes to change on the event other than the attributes send with an event. But we could just make sure that only the state value is used to trigger the event.

Reply:
Added a check on the state as this is a time stamp and only sync the event in case this timestamp has been changed.